### PR TITLE
Refactor GoogleQuotesReader

### PR DESCRIPTION
**Pull request by gliptak**
https://github.com/pydata/pandas-datareader/issues/184
https://github.com/pydata/pandas-datareader/pull/177
@femtotrader

### CODE REVIEW

File: pandas_datareader/tests/test_google.py [hunk 1]
```diff
@@ -56,6 +56,10 @@ def test_google(self):
         self.assertRaises(Exception, web.DataReader, "NON EXISTENT TICKER",
                           'google', start, end)
 
+    def test_get_quote_string(self):
+        df = web.get_quote_google('GOOG')
+        self.assertTrue(df.ix['GOOG']['last'] > 0.0)
+
```
Comment (sinhrks) [inline pandas_datareader/tests/test_google.py]: Pls test ``index`` and ``columns`` are correct using ``tm.assert_index_equal`` also in ``test_get_quote_stringlist``.

Comment (gliptak) [inline pandas_datareader/tests/test_google.py]: Done.

**Comment by femtotrader** [comment]
Thanks @gliptak but you might add some tests to ensure type of returned data.

- `time` is probably a `datetime.time` 
- `change_pct` and `last` a `float` or a `Decimal`

**Comment by sinhrks** [comment]
No, dtypes will be ``float``, ``float``, ``datetime[ns]``. #188 adds a similar dtype test.


Using ``Decimal`` is regarded as ``obejct`` internally, and it causes some numeric ops not work.

**Comment by gliptak** [comment]
Thanks for the pointers. Please review update.

**Comment by gliptak** [comment]
```strptime``` doesn't parse EST with Python 3.x ...
http://bugs.python.org/issue22377 http://stackoverflow.com/questions/14067346/date-format-incorrect-using-strptime

**Comment by gliptak** [comment]
Some code gymnastics ...

**Comment by gliptak** [comment]
Any other changes you would like? Thanks

**Comment by femtotrader** [comment]
That sound good to me. And you @sinhrks ? Can you squash commits @gliptak ?

**Comment by gliptak** [comment]
Squashed.

https://github.com/numpy/numpy/issues/7390

**Comment by femtotrader** [comment]
Thanks @gliptak

**Pull request merged**